ic-agent/src/agent/builder.rs

```diff
@@ -87,4 +87,11 @@ impl AgentBuilder {
         self.config.ingress_expiry = ingress_expiry;
         self
     }
+
+    /// Allows disabling query signature verification. Query signatures improve resilience but require
+    /// a separate read-state call to fetch node keys.
```

Comment:
Should we mention that once a canister -> subnet mapping is looked up, it is cached and never invalidated?  If canisters can change subnets, then this is something that someone who enables verify_query_signatures should be aware of.

Reply:
I had thought it was transitively invalidated by the subnet information being invalidated, since it's just an index, but hadn't considered the possibility of the subnet's ranges changing.

Reply:
I think [subnet splitting](https://github.com/dfinity/portal/pull/1966) would be an event that would change the subnet of some canisters